source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -269,25 +269,21 @@ void Filter::parseV1Header(char* buf, size_t len) {
   }
 }
 
-bool Filter::parseExtensions(os_fd_t fd, uint8_t* buf, size_t buf_size, size_t* buf_off) {
+bool Filter::parseExtensions(Network::IoHandle& io_handle, uint8_t* buf, size_t buf_size,
+                             size_t* buf_off) {
   // If we ever implement extensions elsewhere, be sure to
   // continue to skip and ignore those for LOCAL.
   while (proxy_protocol_header_.value().extensions_length_) {
-    int bytes_avail;
-    auto& os_syscalls = Api::OsSysCallsSingleton::get();
-    if (os_syscalls.ioctl(fd, FIONREAD, &bytes_avail).rc_ < 0) {
```

Comment:
I'm at a loss here: why is it safe to skip these os_syscalls.ioctl calls that we used to have here?

It may make sense to move changes to this file to a different PR.

Reply:
ioctl was used to determine how much data is available for reading but, even ignoring the scenario where data is written after ioctl returns and before recv is called, the result is somewhat redundant because non-blocking fds return EAGAIN when there's no data to read. 

The change is to just call recv with buffer length set to max needed (or max buflen) and if EAGAIN return what we did when ioctl determined there was nothing to read, i.e., false. All other socket errors are treated as exceptions and whatever recv returned (needs to be positive) is what was successfully read. So, we don't really need to know how much is available for reading. 

Will move to separate PR to simplify this one.